account_invoice_ubl/models/account_invoice.py

```diff
@@ -111,14 +146,33 @@ def _ubl_add_legal_monetary_total(self, parent_node, ns, version='2.1'):
             currencyID=cur_name)
         payable_amount.text = '%0.*f' % (prec, self.residual)
 
+    def _ubl_get_invoice_line_price_unit(
+            self, iline):
+        # Use price_subtotal/qty to compute price_unit to be sure
+        # to get a *tax_excluded* price unit
+        # TODO: Ideally, we should get the line untaxed price_unit and declare
+        # the difference as a line discount (to also support modules like
+        # triple discount). The discount would be declared in a line
+        # ChargeAllowance node of type discount
+        dpo = self.env["decimal.precision"]
+        qty_precision = dpo.precision_get("Product Unit of Measure")
+        price_precision = dpo.precision_get("Product Price")
+        qty = iline.quantity
+        if float_is_zero(qty, precision_digits=qty_precision):
+            price_unit = 0.0
+        else:
+            price_unit = float_round(
+                iline.price_subtotal / float(qty),
```

Comment:
Maybe it's a silly question but why not just get [iline.price_unit](https://github.com/OCA/OCB/blob/10.0/addons/account/models/account_invoice.py#L1241) directly here?

Reply:
The line price_unit can be VAT included. Ideally, we should take the price_unit and remove the included VAT from any tax configured as such.
Taking the price_unit also requires to support any discount and declare it in another section.
Currently, it is the price after the discounts. It was already like that and I didn't want to change and add more complexity on this

Reply:
I finally implemented it properly. I compute the VAT excluded unit price and declare the discount in an additional separate section.

The previous implementation was causing rounding issues in some specific amount.